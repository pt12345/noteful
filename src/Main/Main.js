import React, { Component } from 'react';
import ApiContext from '../ApiContext';
import Note from '../Note/Note'


class Main extends Component {

  static contextType = ApiContext

  getNotes = (notes=[], folderId) => {
    if(Object.keys(folderId).length ===0)
      return notes;

    let arr = [];

    for(let i=0;i<notes.length;i++) {

      if(notes[i].folderId === folderId.folderId) {
        arr.push(notes[i])
      }
      
    }

    return arr;
  }


  render() {
    
    const { notes=[] } = this.context
    const folderId = this.props.folderId || {}
    const folderNotes = this.getNotes(notes, folderId);
    return(
      <section>
          <ul>
             {folderNotes.map(note =>
                <li key={note.id}>
                    <Note id={note.id} name={note.name} modified={note.modified} onDeleteNote={this.handleDeleteNote}/>
                </li>
             )}
          </ul>
          <div className='new_note'>
              <span>New Note</span>
          </div>
      </section>
    );
  }
}

export default Main;