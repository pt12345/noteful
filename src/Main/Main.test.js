import React from 'react';
import ReactDOM from 'react-dom';
import Main from './Main';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import reactDom from 'react-dom';

describe('Main', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
  
    ReactDOM.render(<BrowserRouter><Main /></BrowserRouter>, div);
  
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders as expected', () => {
    const main = renderer.create(<BrowserRouter><Main /></BrowserRouter>).toJSON();
    expect(main).toMatchSnapshot();
  })

  it('renders without crashing.. againg', () => {
    const div = document.createElement('div');
    reactDom.render(<BrowserRouter><Main /></BrowserRouter>, div);
    reactDom.unmountComponentAtNode(div);
  })
})
