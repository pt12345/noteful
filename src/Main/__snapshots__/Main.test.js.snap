// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders as expected 1`] = `
<section>
  <ul />
  <ul
    className="new_note"
  >
    <li
      className="new_note_link"
    >
      <a
        aria-current={null}
        href="/folder/addNote"
        onClick={[Function]}
      >
        Add Note
      </a>
    </li>
  </ul>
</section>
`;
